.editorconfig

```
@@ -1,17 +1,350 @@
-# EditorConfig is awesome: https://EditorConfig.org
+### From: https://github.com/dotnet/aspnetcore/blob/main/.editorconfig
+### Last sync: 120b944
+### Disabled: IDE0073
+; EditorConfig to support per-solution formatting.
+; Use the EditorConfig VS add-in to make this work.
+; http://editorconfig.org/
+;
+; Here are some resources for what's supported for .NET/C#
+; https://kent-boogaart.com/blog/editorconfig-reference-for-c-developers
+; https://learn.microsoft.com/en-us/visualstudio/ide/editorconfig-code-style-settings-reference
+;
+; Be **careful** editing this because some of the rules don't support adding a severity level
+; For instance if you change to `dotnet_sort_system_directives_first = true:warning` (adding `:warning`)
+; then the rule will be silently ignored.
 
-# top-most EditorConfig file
+; This is the default for the codeline.
 root = true
 
 [*]
-end_of_line = lf
-insert_final_newline = true
+indent_style = space
+charset = utf-8
 trim_trailing_whitespace = true
+insert_final_newline = true
 
-[*.{cs,vb}]
-tab_width = 4
+[*.cs]
 indent_size = 4
-indent_style = tab
+dotnet_sort_system_directives_first = true
+
+# Don't use this. qualifier
+dotnet_style_qualification_for_field = false:suggestion
+dotnet_style_qualification_for_property = false:suggestion
+
+# use int x = .. over Int32
+dotnet_style_predefined_type_for_locals_parameters_members = true:suggestion
+
+# use int.MaxValue over Int32.MaxValue
+dotnet_style_predefined_type_for_member_access = true:suggestion
+
+# Require var all the time.
+csharp_style_var_for_built_in_types = true:suggestion
+csharp_style_var_when_type_is_apparent = true:suggestion
+csharp_style_var_elsewhere = true:suggestion
```

```suggestion
csharp_style_var_elsewhere = false
```
I'm not entirely sure how the syntax works, but i don't think its useful to have `var` literally everywhere. When type is not clear it should not be `var` and it shouldn't suggest to try change it to `var`.